src/torch_models.py

```diff
@@ -364,26 +384,60 @@ def _initialize_net(self) -> None:
         self._linears.append(nn.Linear(self._hid_sizes[-1], 1))
 
     def _create_loss_fn(self) -> Callable[[Tensor, Tensor], Tensor]:
-        return nn.BCEWithLogitsLoss()
+
+        # See the class docstring for context.
+        def _loss_fn(Y_hat: Tensor, Y: Tensor) -> Tensor:
+            pred = Y_hat.reshape(Y.shape)  # (num data, num classes)
+            log_probs = F.log_softmax(pred / self._temperature, dim=-1)
+            # Note: batchmean is recommended in the PyTorch documentation
+            # and will become the default in a future version.
+            loss = F.kl_div(log_probs, Y, reduction='batchmean')
```

Comment:
i thought the loss would be a multiclass classification loss like you were saying above. is this equivalent?

Reply:
yeah, I think they should be equivalent: https://stats.stackexchange.com/questions/357963/what-is-the-difference-cross-entropy-and-kl-divergence

(I'm opting for KL divergence just in case, because of the reference implementation that I'm following.)

to your second question, I'll add more comments in the code